apps/arweave/src/ar_mining_worker.erl

```diff
@@ -245,281 +277,181 @@ terminate(_Reason, _State) ->
-			update_sub_chunk_cache_size(
-				-(2 * ar_block:get_nonces_per_recall_range(PackingDifficulty)),
-				SessionKey, State)
-	end,
-	{noreply, State3};
-
+%% @doc Handle the `computed_h1` task.
+%% Indicates that the single hash for the first recall range has been computed.
 handle_task({computed_h1, Candidate, _ExtraArgs}, State) ->
-	#mining_candidate{ h1 = H1, chunk1 = Chunk1, session_key = SessionKey } = Candidate,
+	#mining_candidate{ h1 = H1 } = Candidate,
 	State2 = hash_computed(h1, Candidate, State),
-	case h1_passes_diff_checks(H1, Candidate, State2) of
+	State3 = case h1_passes_diff_checks(H1, Candidate, State2) of
 		true ->
+			%% h1 solution found, report it.
 			?LOG_INFO([{event, found_h1_solution},
 				{step, Candidate#mining_candidate.step_number},
 				{worker, State2#state.name},
-				{h1, ar_util:encode(H1)}, 
+				{h1, ar_util:encode(H1)},
 				{p1, Candidate#mining_candidate.partition_number},
 				{difficulty, get_difficulty(State2, Candidate)}]),
-			ar_mining_stats:h1_solution(),
-			%% Decrement 1 for chunk1:
-			%% Since we found a solution we won't need chunk2 (and it will be evicted if
-			%% necessary below)
-			State3 = remove_sub_chunks_from_cache(Candidate, 1, State2),
 			ar_mining_server:prepare_and_post_solution(Candidate),
-			{noreply, State3};
-		Result ->
-			case Result of
-				partial ->
-					ar_mining_server:prepare_and_post_solution(Candidate);
-				_ ->
-					ok
-			end,
-			{ok, Config} = application:get_env(arweave, config),
-			case cycle_sub_chunk_cache(Candidate, {chunk1, Chunk1, H1}, State2) of
-				{cached, State3} ->
-					%% Chunk2 hasn't been read yet, so we cache Chunk1 and wait for
-					%% Chunk2 to be read.
-					{noreply, State3};
-				{do_not_cache, State3} ->
-					%% This node does not store Chunk2. If we're part of a coordinated
-					%% mining set, we can try one of our peers, otherwise we're done.
-					case Config#config.coordinated_mining of
-						false ->
-							ok;
-						true ->
-							DiffPair =
-								case get_partial_difficulty(State3, Candidate) of
-									not_set ->
-										get_difficulty(State3, Candidate);
-									PartialDiffPair ->
-										PartialDiffPair
-								end,
-							ar_coordination:computed_h1(Candidate, DiffPair)
-					end,
-					%% Decrement 1 for chunk1:
-					%% do_not_cache indicates chunk2 was not and will not be read or cached
-					{noreply, update_sub_chunk_cache_size(-1, SessionKey, State3)};
-				{{chunk2, Chunk2}, State3} ->
-					%% Chunk2 has already been read, so we can compute H2 now.
-					ar_mining_hash:compute_h2(
-						self(), Candidate#mining_candidate{ chunk2 = Chunk2 }),
-					%% Decrement 2 for chunk1 and chunk2:
-					%% 1. chunk2 was previously read and cached
-					%% 2. chunk1 that was just read and used to compute H1
-					{noreply, update_sub_chunk_cache_size(-2, SessionKey, State3)}
-			end
+			ar_mining_stats:h1_solution(),
+			%% Update the cache to store h1.
+			case ar_mining_cache:with_cached_value(
```

Comment:
Do we need this here since we're also doing it down here: https://github.com/ArweaveTeam/arweave/pull/697/files#diff-7d9b88c10de9e64afe9594e33a3e801b5b6a9421c6aa51f4d7e529a26a05fa9eR409